ts/modules/full-screen.src.ts

```diff
@@ -18,63 +15,194 @@ import H from '../parts/Globals.js';
+        exportDivElements[menuItems.indexOf('viewFullscreen')].innerHTML = (
+            exportingOptions.menuItemDefinitions.viewFullscreen.text ||
+            chart.options.lang.viewFullscreen
+        );
+    }
+
+    // Unbind event as it's necessary only before exiting from fullscreen.
+    if (chart.unbindFullscreenEvent) {
+        chart.unbindFullscreenEvent();
+    }
+
+    chart.isFullscreen = false;
+};
+
+/* eslint-disable valid-jsdoc */
+
+/**
+ * Exporting module required. Toggles displaying the chart in fullscreen mode.
+ * By default, when the exporting module is enabled, a context button with
+ * a drop down menu in the upper right corner accesses this function.
+ *
+ * @since       next
+ *
+ * @sample      highcharts/members/chart-togglefullscreen/
+ *              Toggle fullscreen mode from a HTML button
+ *
+ * @function    Highcharts.Chart#toggleFullscreen
+ * @return      {void}
+ * @requires    modules/exporting
+ * @requires    modules/fullscreen
+ */
+H.Chart.prototype.toggleFullscreen = function (): void {
+    const chart = this;
+
+    if (!(chart.container.parentNode instanceof HTMLElement)) {
+        return;
+    }
+
+    const container = chart.container.parentNode;
+
+    // Hold event and methods available only for a current browser.
+    if (!chart.browserProps) {
+        if (typeof container.requestFullscreen === 'function') {
+            chart.browserProps = {
+                fullscreenChange: 'fullscreenchange',
+                requestFullscreen: 'requestFullscreen',
+                exitFullscreen: 'exitFullscreen'
+            };
         } else if (container.mozRequestFullScreen) {
-            promise = container.mozRequestFullScreen();
-        } else if (container.webkitRequestFullscreen) {
-            promise = container.webkitRequestFullscreen();
+            chart.browserProps = {
+                fullscreenChange: 'mozfullscreenchange',
+                requestFullscreen: 'mozRequestFullScreen',
+                exitFullscreen: 'mozCancelFullScreen'
+            };
+        } else if (container.webkitRequestFullScreen) {
+            chart.browserProps = {
+                fullscreenChange: 'webkitfullscreenchange',
+                requestFullscreen: 'webkitRequestFullScreen',
+                exitFullscreen: 'webkitExitFullscreen'
+            };
         } else if (container.msRequestFullscreen) {
-            promise = container.msRequestFullscreen();
+            chart.browserProps = {
+                fullscreenChange: 'MSFullscreenChange',
+                requestFullscreen: 'msRequestFullscreen',
+                exitFullscreen: 'msExitFullscreen'
+            };
         }
+    }
 
-        if (promise) {
-            promise['catch']((): void => { // eslint-disable-line dot-notation
-                alert('Full screen is not supported inside a frame'); // eslint-disable-line no-alert
-            });
-        }
+    if (!chart.isFullscreen) {
+        chart.openFullscreen();
```

Comment:
The error message is now removed. Previously I would get an alert when trying to full-screen a chart from inside a frame, now it just silently fails but with a JS error in the console. This can be tested in the utils.